source/agora/script/Signature.d

```diff
@@ -162,6 +170,14 @@ public Hash getChallenge (in Transaction tx, in SigHash sig_hash = SigHash.All,
 
     switch (sig_hash)
     {
+    case SigHash.NoInput:  // eltoo support
+        Transaction dup;
+        // it's ok, we'll dupe the array before modification
+        () @trusted { dup = *cast(Transaction*)&tx; }();
+        dup.inputs = dup.inputs.dup;
+        dup.inputs[input_idx] = Input.init;  // blank out matching input
+        return hashMulti(dup, sig_hash);
```

Comment:
IMHO the proper way to do this would be via a simple wrapper struct, but we can do this later, since you just moved the code.